lib/ex_uid2/api.ex

```diff
@@ -0,0 +1,29 @@
+defmodule ExUid2.Api do
+  alias ExUid2.Api.Request
+  alias ExUid2.Api.Response
+
+  def post_encrypted_request(path, payload, ts \\ :os.system_time(:millisecond)) do
+    secret_key = get_secret_key()
+
+    with {:request_body, {:ok, request_body}} <- {:request_body, Request.make_encrypted_request_body(payload, secret_key, ts)},
```

Comment:
For my own education, why not simply do the following:
```suggestion
    with {:ok, request_body} <- Request.make_encrypted_request_body(payload, secret_key, ts),
```
Is the extra atom wrapped in a tuple there for documentation purposes? Note that the atom is generally repeated in the name of the binded values on the left side so the reader can, with a little more effort, fairly easily deduce what's going on.

Reply:
Ah I think I understand. The extra atom is used in the else statement to extract relevant error messages based on which stage failed.

Reply:
That's exactly right :)